pkg/reconciler/subscription/subscription.go

```diff
@@ -277,18 +287,19 @@ func (r Reconciler) subPresentInChannelSpec(subscription *v1alpha1.Subscription,
 	return false
 }
 
-// Todo: this needs to be changed to use cache rather than a API call each time
 func (r *Reconciler) getChannelable(ctx context.Context, namespace string, chanReference *corev1.ObjectReference) (*eventingduckv1alpha1.Channelable, error) {
-	s, err := eventingduck.ObjectReference(ctx, r.DynamicClientSet, namespace, chanReference)
+	lister, err := r.channelableTracker.ListerFor(*chanReference)
 	if err != nil {
+		logging.FromContext(ctx).Error(fmt.Sprintf("Error getting lister for Channel: %s/%s", namespace, chanReference.Name), zap.Error(err))
```

Comment:
Remove the %s/%s pattern throughout the PR. Prefer using a zap attribute. 

As an example, if this line returns an error, the namespace and name are virtually worthless. We would actually want the GVK (possibly embedded in the error returned by ListerFor, but we shouldn't rely on that). By having the chanReference as a zap attribute, we have that information.

Reply:
Done! thanks!